src/UI/Art/SpriteArtist/CommanderOverlayArtist.java

```diff
@@ -7,108 +7,122 @@
 import java.util.Map;
 
 import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderAbility;
+import Engine.Army;
 import Terrain.GameMap;
 import UI.COStateInfo;
 
 public class CommanderOverlayArtist
 {
-  public static final int OVERLAY_WIDTH = 97;
-  public static final int OVERLAY_HEIGHT = 20;
-
   private static final int animIndexUpdateInterval = 13;
   
   private static Map<String, Sprite> activeAbilityTextSprites = new HashMap<String, Sprite>();
 
   /**
    * Draws an overlay for all COs passed in, left-justified, with status text and optionally highlights one CO
    */
-  public static BufferedImage drawAllCommanderOverlays(Commander[] coList, GameMap drawableMap, int drawingWidth, int drawingHeight, Commander coToHighlight)
+  public static BufferedImage drawAllCommanderOverlays(Army[] armyList, GameMap drawableMap, int drawingWidth, int drawingHeight, Army armyToHighlight)
   {
     BufferedImage output = SpriteLibrary.createTransparentSprite(drawingWidth, drawingHeight);
     Graphics g = output.getGraphics();
     int verticalOffset = 0;
 
-    for( Commander CO : coList )
+    for( Army army : armyList )
     {
+      BufferedImage headerWorkspace = SpriteLibrary.createTransparentSprite(drawingWidth, drawingHeight);
+      int newOffset = CommanderOverlayArtist.drawCommanderOverlay(headerWorkspace.getGraphics(), army, verticalOffset, true);
```

Comment:
This is creating a new sprite the size of the entire canvas for each commander overlay. Then you are drawing to this canvas in the correct absolute offset position, and drawing this image to the original canvas at 0, 0.

Why do we need to double-buffer this? It seems wildly inefficient.

Reply:
`drawAllCommanderOverlays()` wants to draw the highlight underneath the actual header.
It doesn't know how big any given *Army* overlay will be, since I didn't initially have hooks to calculate that "for real" before, and didn't get around to revising.
So it draws the overlay first, gets how big it is, and uses that for the highlight before committing the actual drawn header.

I will take a shot at fixing this on `presentationFixes`.